### PR TITLE
More semconv features and cleanups

### LINKED ISSUE #84: Cannot run `weaver-forge` tests on Windows
Today, there are two major things blocking weaver-forge tests running on windows:

1. Test templates output file path names in unix-style without forcing this style via rust code or rendering filter.
2. Test harness should ignore line endings ("\r\n" vs "\n") in windows *or* the template engine should not use platform-specific line endings.

### PR DESCRIPTION
**Pull request by jsuereth**
- Add new helper filter to split group ids into constituent parts, e.g. `"registry.foo" => ["registry", "foo"]`
- Update weaver_forge tests to work on windows
  - Translate"\r\n" => "\n" for compared output
  - Add filename filter to jinja templates that converts "\" to "/"
  - Improve error output when file system comparison fails.
  - Add 'trim' to a few templates to help remove platform specific EOLs
- Update Dockerfile to use debug mode, so we get debug output on template failure

Fixes #84

### CODE REVIEW

**Comment by jsuereth** [comment]
Note: Overall weaver on windows tests still fail in other crates, mostly anything that prints file paths with expected output files.

**Comment by lquerel** [comment]
> Note: Overall weaver on windows tests still fail in other crates, mostly anything that prints file paths with expected output files.

I don't have access to Windows, unfortunately. Can you create a GH issue to describe what's causing problems on Windows? It could be a perfect issue for someone who wants to start contributing to this repo.

File: Dockerfile [hunk 1]
```diff
@@ -11,11 +11,12 @@ COPY data /build/data
 COPY src /build/src
 COPY templates /build/templates
 COPY tests build/tests
-RUN cargo build --release
+# Don't build release, so we get template debugging output.
```
[the remark this answers is not in the document]
Comment (jsuereth) [reply Dockerfile:14]: I can open a bug to fix this going forward.  So far two major features of JINJA are missing:


1. `debug(...)` method does not output on template.
2. Template failure messages are NOT rendered, but squashed.  In debug builds, you get a nice message about *where* a template file failed and what context was sent to the template (allowing you to debug jq queries).

This is likely just a config around print statements or some such, but I didn't have time to dive into the details.

[the remark this answers is not in the document]
Comment (jsuereth) [reply Dockerfile:14]: I opened a bug for this: #98

File: crates/weaver_forge/templates/test/attribute_group.md [hunk 2]
```diff
@@ -1,7 +1,7 @@
 {%- set file_name = ctx.id | file_name -%}
 {{- template.set_file_name("attribute_group/" ~ file_name ~ ".md") -}}
 
-## Group `{{ ctx.id }}` ({{ ctx.type }})
+## Group `{{ ctx.id | split_id | list | join("_") }}` ({{ ctx.type }})
```
[the remark this answers is not in the document]
Comment (jsuereth) [reply crates/weaver_forge/templates/test/attribute_group.md:4]: Moved to vec.  Wierdly minijinja still is confused on values.

[the remark this answers is not in the document]
Comment (lquerel) [reply crates/weaver_forge/templates/test/attribute_group.md:4]: Based on the code of the list filter (see below) you need to wrap the vec into a value.

```rust
#[cfg_attr(docsrs, doc(cfg(feature = "builtins")))]
pub fn list(state: &State, value: Value) -> Result<Value, Error> {
    let iter = ok!(state.undefined_behavior().try_iter(value).map_err(|err| {
        Error::new(ErrorKind::InvalidOperation, "cannot convert value to list").with_source(err)
    }));
    Ok(Value::from(iter.collect::<Vec<_>>()))
}
```

File: crates/weaver_forge/src/lib.rs [hunk 3]
```diff
@@ -421,6 +422,23 @@ fn flatten(value: Value) -> Result<Value, minijinja::Error> {
...
 
+// Helper function to take an "id" and split it by '.' into namespaces.
+fn split_id(value: Value) -> Result<Vec<Value>, minijinja::Error> {
+    match value.as_str() {
+        Some(id) => {
+            let values: Vec<Value> = id
+                .split('.')
+                .map(|s| Value::from_safe_string(s.to_owned()))
+                .collect();
+            Ok(values)
```
Comment (lquerel) [inline crates/weaver_forge/src/lib.rs:433]:
```suggestion
            Ok(Value::from(values))
```

Comment (jsuereth) [reply crates/weaver_forge/src/lib.rs]: FYI - merged this suggestion and it breaks everything.

Comment (lquerel) [reply crates/weaver_forge/src/lib.rs:433]: Ok

**Review by lquerel — APPROVED** [review]

**Pull request merged**